src/java.base/share/classes/java/lang/Class.java

```diff
@@ -923,6 +923,13 @@ public boolean isSynthetic() {
      *     returns "[[[[[[[I"
      * </pre></blockquote>
      *
+     * @apiNote
+     * This is not a strict inverse of {@link #forName(String) forName}.
+     * For normal classes, distinct classes which share a common name but
```

Comment:
Readers may wonder what "normal classes" means here, it might be simpler to just drop introducing that term and just use the structure from the note removed from descriptorString.

Reply:
This might wish to refer to "classes that can be described nominally", but same might be said for hidden classes as well. So your suggestion is better.

Reply:
On a second thought, these are all `getName` results that cannot be parsed by `forName`:
 - Primitive type names
 - Hidden class names
 - Arrays whose component types are hidden classes

I suggest:
```
     * This is not a strict inverse of {@link #forName(String) forName(String)}.
     * Distinct classes which share a common name but have different class
     * loaders will have the identical name. {@linkplain #isPrimitive()
     * Primitive types}, {@linkplain #isHidden() hidden} classes, and arrays
     * with hidden class as element types cannot be discovered by {@code forName}.
```

Reply:
That reads fine with one minor tweak:

... and array classes whose element type is a hidden class or interface cannot be discovered by {@code forName}.